lib/spack/spack/util/windows_registry.py

```diff
@@ -257,12 +300,51 @@ def find_subkey(self, subkey_name, recursive=True):
         Return:
             the desired subkey as a RegistryKey object, or none
         """
+        return self._find_subkey_s(
+            subkey_name,
+            WindowsRegistryView.KeyMatchConditions.name_matcher(subkey_name),
+            recursive=recursive,
+        )
 
-        if not recursive:
-            return self.get_subkey(subkey_name)
+    def find_matching_subkey(self, subkey_name, recursive=True):
+        """If non recursive, this method is the same as get subkey with error handling
+        Otherwise perform a BFS of subkeys until a key matching subkey name regex is found
+        Returns None or the first RegistryKey object corresponding to requested key name
 
-        else:
-            return self._traverse_subkeys(lambda x: x.name == subkey_name)
+        Args:
+            subkey_name (str): regex string representing subkey to be searched for
+            recursive (bool): optional argument, if True, subkey need not be a direct
+                                sub key of this registry entry, and this method will
+                                search all subkeys recursively.
+                                Default is True
+        Return:
+            the desired subkey as a RegistryKey object, or none
+        """
+        return self._find_subkey_s(
+            subkey_name,
+            WindowsRegistryView.KeyMatchConditions.regex_matcher(subkey_name),
+            recursive=recursive,
+        )
+
+    def find_subkeys(self, subkey_name, recursive=True):
+        """Exactly the same as find_subkey, except this function tries to match
+        a regex to multiple keys
+
+        Args:
+            subkey_name (str): string representing subkey to be searched for
```

Comment:
Since `_find_subkey_s` is a private method, I think I'd like the docstrings of the public methods exposed in full, which is particularly useful for developers using tools like VSCode or the `help` method so they don't have to navigate to a second, private method for more details, which feels contrary to the point of a private method to me.

Reply:
I get your point, but duplicating the docstring leads to its own errors (e.g. I'm guessing this said "string" vs. "regex string" is because the docstring was copy-pasted, and future docstring edits could likewise lead to discrepancies), and these methods *do* in fact depend entirely on the underlying method. The summary string *should* remain for each method; perhaps it would be sufficient to remove the longer argument descriptions - just identify the type. 

There is another spot where several methods refer to one IIRC, and in that case we create a module-level docstring variable that is reused across all those methods, so that would also work here.